File: Python/1.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type target: int
              nums: List[int]
        :rtype: List[int]
        """
        d = {}
        for i in range(len(nums)):
            if nums[i] in d:
                return [i, d[nums[i]]]
            else:
                d[target - nums[i]] = i

File: C++/1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        """
        :type: nums: vector<int>
        :type: target: int
        :rtype: array
        """
        unordered_map<int, int> d;
        for (int i = 0; i < nums.size(); i++) {
            if (d.count(nums[i]) == 1)
                return {d[nums[i]], i};
            else
                d[target-nums[i]] = i;
        }
        return {};
    }
};